src/reference/snakemake-style-guide.rst

```diff
@@ -61,3 +61,122 @@ global variable. 3 ways are supported, but only 2 should be used:
 3. ``config.get(key)``: Never use this. All use cases should be covered
    by (1) and (2). Using this will only mask errors that may be due to a
    missing required key.
+
+Use a config.yaml file
+======================
+
+Configuration is data and should live inside a YAML file named
+``config.yaml``. You can access it in your Snakefile by including the
+line:
+
+.. code-block:: python
+
+   configfile: "config.yaml"
+
+and then using the ``config`` dictionary provided in scope afterwards.
+
+Allow for easy config overrides
+===============================
+
+By including the following snippet in your Snakefile, you can allow for
+optional configuration overrides using the ``--configfile`` option to
+``snakemake``.
+
+.. code-block:: python
+
+   if not config:
+       configfile: "config.yaml"
```

Comment:
These recommendations seems at odds with each other. Which should it be?

Reply:
This is a good point. I'm not sure since [ncov](https://github.com/nextstrain/ncov/blob/d17782f5ea422052c6ec6d2cc03ec8346ebdee0c/Snakefile#L38) uses the former and [monkeypox](https://github.com/nextstrain/monkeypox/blob/12ccb428aabccd9271fe8c5e61533edd656474e4/Snakefile#L14-L16) uses the latter.

If I had to choose, I would go with the former because it allows for partial overrides through Snakemake CLI parameters. The latter requires `config` to be all-or-nothing.

I will try creating a PR in the monkeypox repo using the former approach to see if it makes sense to recommend that here.

Reply:
I agree that we should use the first example in the snippet above, since that allows the entire config file to be overridden from the command line with `--configfile` and it allows specific config variables from the default config to be overridden from the command line.

Checking for the presence of an existing global config like the second block in the snippet is something we might do in a workflow-specific manner especially for workflows that allow users to define custom rules, but I wouldn't recommend that as the standard default behavior.

Reply:
Updated in 7fb0ecbfdd312a3f10ca531510092fa1066e7f77. Applied to the monkeypox repo: https://github.com/nextstrain/monkeypox/pull/173